### PR TITLE
Switch relationship direction from parent FileNode to FileNode and rename from 'parent' to 'child'

### PR DESCRIPTION
**Pull request by elisabeth-sorrell**
Refer to https://github.com/SeedCompany/cord-api-v3/issues/1753 for more info. This help make sure all BaseNode to BaseNodes are the same relationship direction.

┆Issue is synchronized with this [Monday item](https://seed-company-squad.monday.com/boards/3451697530/pulses/3466553805) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by elisabeth-sorrell** [comment]
Closes https://github.com/SeedCompany/cord-api-v3/issues/2424

**Review by CarsonF — COMMENTED** [review]
Previously migrations should not be changed as they would be working with the old schema.

**Review by CarsonF — CHANGES_REQUESTED** [review]
See prev comment

File: src/components/file/file.repository.ts [hunk 1]
```diff
@@ -109,11 +109,11 @@ export class FileRepository extends CommonRepository {
       .query()
       .match([
         node('start', 'FileNode', { id }),
-        relation('out', 'parent', 'parent', ACTIVE, '*'),
+        relation('in', 'child', 'child', ACTIVE, '*'),
```
Comment (CarsonF) [inline src/components/file/file.repository.ts:112]: I think the query variable should continue to be called `parent` as it is one of the _parents_ of the node.
```suggestion
        relation('in', 'parent', 'child', ACTIVE, '*'),
```

File: src/components/file/migrations/ChangeFileToFileRels.migration.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      .query()
+      .match([
+        node('parent', 'FileNode'),
+        relation('out', 'p', 'parent'),
+        node('child'),
+      ])
+      .raw(
+        `
+        with collect(p) AS rels
+        CALL apoc.refactor.rename.type("parent", "child", rels )
```
Comment (CarsonF) [inline src/components/file/migrations/ChangeFileToFileRels.migration.ts:33]: Can we combine these two queries? Either one after the other or maybe both in sub queries after matching the appropriate relationships?

File: src/components/file/migrations/ChangeFileToFileRels.migration.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  async up() {
+    // first, invert the relationship from incoming to outgoing
+    const res = await this.db
+      .query()
+      .match([
+        node('parent', 'FileNode'),
+        relation('in', 'rel', 'parent'),
+        node('child'),
+      ])
+      .raw(`CALL apoc.refactor.invert(rel) yield input, output`)
```
Comment (CarsonF) [inline src/components/file/migrations/ChangeFileToFileRels.migration.ts:15]: We also have around 500k of these relationships in prod. Any idea if either of these procedures will handle this? Or do we need to wrap them in an `apoc.periodic.iterate`?

**Comment by elisabeth-sorrell** [comment]
@CarsonF back to you

**Comment by elisabeth-sorrell** [comment]
@CarsonF Just making sure you know that the comments on the PR have been addressed

**Comment by sethmcknight** [comment]
Too old

**Pull request closed without merging**